### PR TITLE
Update user guide for changing the save location

### LINKED ISSUE #61: Update user guide for changing the save location
User guide for changing of the save location needs to be updated to the new implementation

### PR DESCRIPTION
**Pull request by brandonyeoxg**
Fixes #61 

Update user guide to reflect the new behaviour of isValidFilePath

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,10 +104,14 @@ There is no need to save manually.
...
-Example: 
+Example:
 * `java seedu.addressbook.AddressBook mydata.txt`
+* `java seedu.addressbook.AddressBook myFolder/mydata.txt`
 
-> The file name must end in `.txt` for it to be acceptable to the program.
+> The file path must contain a valid file name and a valid parent directory.
+> File name is valid if it has an extension and no reserved characters(OS-dependent).
+> Parent directory is valid if the parent directory exists.
+> If a file already exists, it must be a regular file.
```
Comment (pyokagan) [inline README.md]: If you look at the markdown rendering, this whole block is squished together in one line, which may not be what you were intending?

File: README.md [hunk 2]
```diff
@@ -104,10 +104,14 @@ There is no need to save manually.
...
 You can change the location by specifying the file path as a program argument.
 
-Example: 
+Example:
 * `java seedu.addressbook.AddressBook mydata.txt`
+* `java seedu.addressbook.AddressBook myFolder/mydata.txt`
 
-> The file name must end in `.txt` for it to be acceptable to the program.
+> The file path must contain a valid file name and a valid parent directory.
+> File name is valid if it has an extension and no reserved characters(OS-dependent).
```
Comment (pyokagan) [inline README.md]: `s/characters(OS/characters (OS/`

File: README.md [hunk 3]
```diff
@@ -104,10 +104,14 @@ There is no need to save manually.
...
+> Parent directory is valid if the parent directory exists.
```
Comment (pyokagan) [inline README.md]: `Parent directory is valid if it exists.`

**Comment by brandonyeoxg** [comment]
Oops please ignore the latest commits. I created a separate branch to test markdown formatting.

**Comment by brandonyeoxg** [comment]
@pyokagan Apologies for the user guide formatting mistake.

File: README.md [hunk 4]
```diff
@@ -104,12 +104,16 @@ There is no need to save manually.
 Address book data are saved in a file called `addressbook.txt` in the project root folder.
 You can change the location by specifying the file path as a program argument.
 
-Example: 
+Example:
 * `java seedu.addressbook.AddressBook mydata.txt`
```
Comment (pyokagan) [inline README.md]: Add a newline above this line otherwise the Markdown won't render correctly on GitHub pages.

**Comment by pyokagan** [comment]
@damithc Do you have any comments on the wording of the documentation?

**Comment by damithc** [comment]
Looks mostly fine, but I'm not sure what we mean by `regular file`.

**Comment by pyokagan** [comment]
> Looks mostly fine, but I'm not sure what we mean by regular file.

Well the problem is that unix has all kinds of files (and we only accept one type) so it may be imprecise if we used the general term "file".

Any ideas?

**Comment by damithc** [comment]
Something like this?
> Note for non-Windows users: only 'regular' files are acceptable

File: README.md [hunk 5]
```diff
@@ -105,13 +105,14 @@ Address book data are saved in a file called `addressbook.txt` in the project ro
...
 Example:
+
 * `java seedu.addressbook.AddressBook mydata.txt`
 * `java seedu.addressbook.AddressBook myFolder/mydata.txt`
 
 > The file path must contain a valid file name and a valid parent directory.<br>
   File name is valid if it has an extension and no reserved characters (OS-dependent).<br>
   Parent directory is valid if it exists.<br>
-  If a file already exists, it must be a regular file.<br>
+  Note for non-Windows users: only 'regular' files are acceptable.<br>
```
Comment (damithc) [inline README.md]: `Note for non-Windows users: if the file already exists, it must be a 'regular' file.`?

Comment (brandonyeoxg) [inline README.md]: Oh ok got it.

**Comment by pyokagan** [comment]
@damithc Can merge?

**Comment by damithc** [comment]
Yes. can merge. 👍

**Pull request merged**